client/src/home/edit_form/EditForm.js

```diff
@@ -0,0 +1,201 @@
+                <Form.Text className="warning">Item name is required</Form.Text>
+              )}
+
+            </Form.Group>
+  
+            <Form.Group>
+              <Form.Label> Batch ID</Form.Label>
+              <Form.Control
+                type="text"
+                placeholder="e.g. 1234"
+                
+                onBlur={(e) => {
+                  if(newBatchId.length === 4){
+                    setbatchIdWarning(false);
+                  }
+                  else{
+                    setbatchIdWarning(true);
+                  }
+                }}
+                onChange={(event) => {
+                  let val = event.target.value.trim().toLowerCase();
+                  if(val.length === 4){
+                    setbatchIdWarning(false);
+                  }
+                  else{
+                    setbatchIdWarning(true);
+                  }
+                  setOldBatchId(val);
+                  setNewBatchId(val);
+                }}
+              />
+              {showbatchIdWarning && (
+                <Form.Text className="warning">4 digit Batch ID is required</Form.Text>
+              )}
+            </Form.Group>
+  
+            <Form.Group>
+              <Form.Label>Weight (lbs)</Form.Label>
+              <Form.Control
+                type="number"
+                placeholder="e.g. 50"
+                onBlur={(e) => {
+                  if(weight > 0 && weight < Infinity){
+                    
+                    setWeightWarning(false);
+                  }
+                  else{
+                    setWeightWarning(true);
+                  }
+                }}
+                onChange={(event) => {
+                  let val = Number(event.target.value.trim());
+  
+                  // Checks for valid integer
+                  if (val !== Infinity && val > 0) {
+                    setWeight(val);
+                    setWeightWarning(false);
+                  } else {
+                    setWeightWarning(true);
+                  }
+                }}
+              />
+              {showWeightWarning && (
+                <Form.Text className="warning">Invalid weight value</Form.Text>
+              )}
+            </Form.Group>
+  
+            <Form.Group>
+              <Form.Label> Out by date</Form.Label>
+              <DatePicker
+                selected={outByDate}
+                onChange={(date) => setOutByDate(date)}
+              />
+            </Form.Group>
+  
+            
+          </Form.Group>
+          
+          <Button variant="light" onClick={() => props.closeForm()}>
+            Cancel
+          </Button>
+          <Button
+            variant="primary"
+            
+              onClick={() => {
+                if(itemName.length > 0 && newBatchId.length === 4 && weight > 0 && weight < Infinity){
+                    editItem(cookies.token, itemName, oldBatchId, newBatchId, weight, outByDate);
```

Comment:
Is itemName, oldBatchId, newBatchId, weight, outByDate set to whatever original value they were? For example, if the user only decides to changed weight, when you call editItem, are all of the other parameters going to be the value that they originally were? It looks like they will all be "" since in the beginning you set them to be "".

Reply:
Yeah, it will be set to the original values of the batch that the user clicked on. These were temporary implementation that we did because we thought we couldn't really do it without the calendar set up, but we can definitely fix it to have a placeholder batch passed in as prop and fix this too!